storage/StorageExample/Shared/ImagePicker.swift

```diff
@@ -13,119 +13,193 @@
+                if FileManager.default.fileExists(atPath: destURL.path) {
+                  try FileManager.default.removeItem(at: destURL)
+                }
+                try FileManager.default.copyItem(at: url!, to: destURL)
+                continuation.resume()
+              } catch {
+                print("Cannot copy item at \(url!) to \(destURL): \(error)")
+                continuation.resume(throwing: error)
               }
-              try FileManager.default.copyItem(at: url!, to: destURL)
-              continuation.resume()
-            } catch {
-              print("Cannot copy item at \(url!) to \(destURL): \(error)")
             }
           }
         }
+      } catch {
+        print("Cannot load file from the image picker.")
+        throw error
+      }
+    }
+
+    func loadImage() async throws -> UIImage? {
+      do {
+        return try await withCheckedThrowingContinuation { continuation in
+          self.loadObject(ofClass: UIImage.self) { image, error in
+            if let error = error {
+              continuation.resume(throwing: error)
+            }
+            continuation.resume(returning: image as? UIImage)
+          }
+        }
+      } catch {
+        print("Image was not properly loaded.")
+        throw error
       }
-    } catch {
-      print("Cannot load file from the image picker.")
     }
   }
 
-  func loadImage() async throws -> UIImage? {
-    do {
-      return try await withCheckedThrowingContinuation { continuation in
-        self.loadObject(ofClass: UIImage.self) { image, error in
-          if let error = error {
-            continuation.resume(throwing: error)
+#elseif os(macOS)
+
+  struct ImagePicker: View {
+    @Binding var image: NSImage?
+    var imageURL: URL?
+
+    var body: some View {
+      ZStack {
+        if let image = self.image {
+          Image(nsImage: image)
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+        } else {
+          ZStack {
+            Image(systemName: "photo.fill")
+              .resizable()
+              .scaledToFit()
+              .opacity(0.6)
+              .frame(width: 300, height: 200, alignment: .top)
+              .cornerRadius(16)
+              .padding(.horizontal)
+            Text("Drag and drop an image file")
+              .frame(width: 300)
           }
-          continuation.resume(returning: image as? UIImage)
         }
       }
-    } catch {
-      print("Image was not properly loaded.")
-      return nil
+      .frame(height: 320)
+      .background(Color.black.opacity(0.6))
+      .cornerRadius(16)
+      .onDrop(of: ["public.url", "public.file-url"], isTargeted: nil) { (items) -> Bool in
+        if let item = items.first {
+          if let identifier = item.registeredTypeIdentifiers.first {
+            if identifier == "public.url" || identifier == "public.file-url" {
+              item.loadItem(forTypeIdentifier: identifier, options: nil) { urlData, error in
```

Comment:
Curious, does an async/await API generate for this?
```swift
try await item.loadItem( //...
```

Reply:
It looks like [loadItem](https://developer.apple.com/documentation/foundation/nsitemprovider/1403900-loaditem) does get an async note in its doc. Will update it.